updated naming format in py_tester.py

wkukka1 opened the pull request:
changed python test names to appear as [filename] testname.

![image](https://github.com/user-attachments/assets/6a96f605-9c70-4403-ba1c-e89c623040eb)
results from running the autotest_py tests^

wkukka1 commented:
Thanks for the feedback!

For the unit tests should the format be [filename] classname.funcname

david-yz-liu commented:
> For the unit tests should the format be [filename] classname.funcname

Yes that's good 👍

wkukka1 commented:
Updates: 
- updated jupyter notebook name to follow [filename] funcname
- created unit test for jupyter naming conventions
- updated the ci workflow to install required packages for jupyter tester to be used

![image](https://github.com/user-attachments/assets/a41227aa-35c3-4e13-8b8c-4fde30d2683a)
^ updated names
